src/Permeability/Permeability.jl

```diff
@@ -0,0 +1,198 @@
+module Permeability
+
+using Parameters, Unitful, LaTeXStrings, MuladdMacro
+using ..Units
+using ..PhaseDiagrams
+using GeoParams: AbstractMaterialParam, AbstractMaterialParamsStruct, @extractors, add_extractor_functions
+using GeoParams: fastpow, pow_check, @pow
+import ..Units: isdimensional
+using ..MaterialParameters: No_MaterialParam, MaterialParamsInfo
+import Base.show, GeoParams.param_info
+
+include("../Computations.jl")
+
+abstract type AbstractPermeability{T} <: AbstractMaterialParam end
+
+export compute_permeability,     # calculation routines
+    compute_permeability!,       # in place calculation
+    compute_permeability_ratio, # calculation with phase ratios
+    param_info,                  # info about the parameters
+    AbstractPermeability,
+    ConstantPermeability,        # constant
+    HazenPermeability,           # Hazen equation
+    PowerLawPermeability,        # Power-law permeability
+    CarmanKozenyPermeability    # Carman-Kozeny permeability
+
+# Define "empty" computational routines in case nothing is defined
+function compute_permeability!(
+    k::_T, s::No_MaterialParam{_T}; ϕ::_T=zero(_T)
+) where {_T}
+    return zero(_T)
+end
+function compute_permeability(s::No_MaterialParam{_T}; ϕ::_T=zero(_T)) where {_T}
+    return zero(_T)
+end
+
+# Constant Permeability
+"""
+    ConstantPermeability(k = 1e-12m^2)
+
+Defines a constant permeability value for a given material.
+
+# Arguments
+- `k::Float64`: The permeability value in square meters (m^2). Default is `1e-12 m^2`.
+
+# Example
+```julia
+rheology = SetMaterialParams(;
+                      Phase=1,
+                      CreepLaws=(PowerlawViscous(), LinearViscous(; η=1e21Pa * s)),
+                      Gravity=ConstantGravity(; g=9.81.0m / s^2),
+                      Density= MeltDependent_Density(),
+                      Permeability = ConstantPermeability(; k=1e-12m^2),
+                      )
+```
+"""
+@with_kw_noshow struct ConstantPermeability{_T,U} <: AbstractPermeability{_T}
+    k::GeoUnit{_T,U} = 1e-12m^2 # permeability
+end
+ConstantPermeability(args...) = ConstantPermeability(convert.(GeoUnit, args)...)
+isdimensional(s::ConstantPermeability) = isdimensional(s.k)
+
+@inline (s::ConstantPermeability)(; args...) = s.k.val
+@inline (s::ConstantPermeability)(args) = s(; args...)
+@inline compute_permeability(s::ConstantPermeability{_T}, args) where {_T} = s(; args...)
+@inline compute_permeability(s::ConstantPermeability{_T})       where {_T} = s()
+
+function param_info(s::ConstantPermeability)
+    return MaterialParamsInfo(; Equation=L"k = cst")
+end
+
+function compute_permeability!(k::AbstractArray, s::ConstantPermeability; kwargs...)
+    @unpack_val k_val = s
+    k[:] .= k_val
```

Comment:
```suggestion
    k .= k_val
```
Not sure why you want to convert it into a vector.

Reply:
this has also been done [here](https://github.com/JuliaGeodynamics/GeoParams.jl/blob/746ea967046a5b4fad414c8191df741f511cf486/src/Density/Density.jl#L72). This can be removed in both cases I guess

Reply:
Yes indeed